openapi.yaml

```diff
@@ -25,105 +32,552 @@ security:
+            $ref: '#/definitions/PizzaResult'
+          headers:
+            location:
+              type: string
+              description: a link to the created resource
         '400':
           $ref: '#/responses/400'
         '409':
-          description: Conflict
+          $ref: '#/responses/409'
+        '500':
+          $ref: '#/responses/500'
+  /pizzas/{pizzaId}:
+    get:
+      description: Get a pizza with Id pizzaId
+      tags:
+        - pizza
+      parameters:
+        - $ref: '#/parameters/pizzaId'
+      responses:
+        '200':
+          description: A single pizza
           schema:
-            $ref: '#/definitions/ErrorResult'
+            $ref: '#/definitions/PizzaResult'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
+    patch:
+      description: Update a pizza recipe
+      tags:
+        - pizza
+      parameters:
+        - $ref: '#/parameters/pizzaId'
+        - in: body
+          name: attributes
+          description: attributes of the pizza to be updated
+          schema:
+            $ref: '#/definitions/PizzaPatchBody'
+      responses:
+        '200':
+          description: The updated pizza recipe
+          schema:
+            $ref: '#/definitions/PizzaResult'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '409':
+          $ref: '#/responses/409'
+        '500':
+          $ref: '#/responses/500'
+    delete:
+      description: Delete a pizza recipe
+      tags:
+        - pizza
+      parameters:
+        - $ref: '#/parameters/pizzaId'
+      responses:
+        '204':
+          description: successfully deleted the pizza
+        '404':
+          $ref: '#/responses/404'
         '500':
           $ref: '#/responses/500'
-  /pets/{id}:
+  /pizzas/{pizzaId}/relationships/dough:
     get:
-      summary: Info for a specific pet
+      description: Get the dough recipe for a pizza
       tags:
-        - pets
-      description: Get pet by unique ID
-      operationId: getPetById
+        - pizza
+        - dough
       parameters:
-        - name: id
-          in: path
-          description: Unique Pet ID, UUID format
-          required: true
+        - $ref: '#/parameters/pizzaId'
+      responses:
+        '200':
+          description: A dough recipe resource reference
+          schema:
+            $ref: '#/definitions/DoughRelationshipResourceSingleton'
+        '400':
+          $ref: '#/responses/400'
+        '404':
+          $ref: '#/responses/404'
+        '500':
+          $ref: '#/responses/500'
+    patch:
+      description: update the dough relationship for the pizza. Pass `null` as the data to remove.
+      tags:
+        - pizza
+        - dough
+      parameters:
+        - $ref: '#/parameters/pizzaId'
+        - in: body
+          name: dough
+          schema:
+            $ref: '#/definitions/DoughRelationshipResourceSingleton'
```

Comment:
In this request body, I think `data` needs to be required. Could you also fix this for the other PATCH relationships endpoints?